pkg/conn/basedb.go

```diff
@@ -67,6 +68,8 @@ func (d *defaultDBProvider) Apply(config config.DBConfig) (*BaseDB, error) {
 type BaseDB struct {
 	DB *sql.DB
 
+	lock *sync.Mutex
```

Comment:
we often use `sync.Mutex` directly without _pointer_ (`*sync.Mutex`), like https://github.com/golang/go/blob/0f7b4e72a054f974489d8342cdae5a6a7ba7a31b/src/database/sql/sql.go#L413.

I think we are overusing _pointer_.

add some comments to explain the purpose of this `sync.Mutex`? or we can give it a meaningful name.